pkg/apis/ingress/v1alpha1/status.go

```diff
@@ -0,0 +1,190 @@
+	condition := &operatorapi.OperatorCondition{
+		Type:   operatorapi.OperatorStatusTypeSyncSuccessful,
+		Status: operatorapi.ConditionTrue,
+	}
+
+	if err != nil {
+		condition.Status = operatorapi.ConditionFalse
+		condition.Reason = "SyncFailed"
+		condition.Message = err.Error()
+	}
+
+	ci.setStatusCondition(condition)
+}
+
+// SetStatusAvailableCondition modifies the ClusterIngress object's status by
+// setting its Available condition to the specified status and message.
+func (ci *ClusterIngress) SetStatusAvailableCondition(available bool, message string) {
+	condition := &operatorapi.OperatorCondition{
+		Type:   operatorapi.OperatorStatusTypeAvailable,
+		Status: operatorapi.ConditionTrue,
+	}
+
+	if !available {
+		condition.Status = operatorapi.ConditionFalse
+		condition.Reason = "ServiceNotReady"
+		condition.Message = message
+	}
+
+	ci.setStatusCondition(condition)
+}
+
+// clusterIngressStatusEqual returns true if and only if the status conditions
+// (ignoring LastTransitionTime) and ingresses of the provided ClusterIngress
+// objects are equal.
+func clusterIngressStatusEqual(oldCi, ci *ClusterIngress) bool {
+	numConditions := len(ci.Status.Conditions)
+	if numConditions != len(oldCi.Status.Conditions) {
+		return false
+	}
+	for i := 0; i < numConditions; i++ {
+		conditionA := oldCi.Status.Conditions[i]
+		conditionB := ci.Status.Conditions[i]
+
+		// Compare every field except LastTransitionTime.
+		if conditionA.Type != conditionB.Type ||
+			conditionA.Status != conditionB.Status ||
+			conditionA.Reason != conditionB.Reason ||
+			conditionA.Message != conditionB.Reason {
+			return false
+		}
+	}
+
+	numIngresses := len(ci.Status.LoadBalancer.Ingress)
+	if numIngresses != len(oldCi.Status.LoadBalancer.Ingress) {
+		return false
+	}
+	for i := 0; i < numIngresses; i++ {
+		ingressA := oldCi.Status.LoadBalancer.Ingress[i]
+		ingressB := ci.Status.LoadBalancer.Ingress[i]
+
+		if ingressA != ingressB {
+			return false
+		}
+	}
+
+	return true
+}
+
+// UpdateStatus updates the receiver ClusterIngress object's status if it
+// differs from the provided ClusterIngress object's status.
+func (ci *ClusterIngress) UpdateStatus(oldCi *ClusterIngress) error {
+	if clusterIngressStatusEqual(oldCi, ci) {
+		return nil
+	}
+
+	if err := sdk.Update(ci); err != nil {
+		return fmt.Errorf(
+			"failed to update status of ClusterIngress %q: %v",
+			ci.Name, err)
+	}
+
+	return nil
+}
+
+// CreateOrUpdateOperatorStatus creates or updates the OperatorStatus object for
+// the ClusterIngress by setting the specified condition.
+func (ci *ClusterIngress) CreateOrUpdateOperatorStatus(condition osv1.OperatorStatusCondition) error {
+	operatorNamespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		logrus.Fatalf("Failed to get watch namespace: %v", err)
+	}
+
+	operatorName, err := k8sutil.GetOperatorName()
+	if err != nil {
+		logrus.Fatalf("Failed to get operator name: %v", err)
+	}
```

Comment:
I added the fields to `Handler`.  Since this means `CreateOrUpdateOperatorStatus` needs the Handler, and since it didn't really need the ClusterIngress, I changed its receiver to `h *Handler`, moved it to `handler.go`, and changed it to `createOrUpdateOperatorStatus`.